environment/create_google_config.py

```diff
@@ -0,0 +1,308 @@
+
+    	print "\nEnter the \"Billing Account ID\" of the billing account you want to use\nto create a new Google project."
+    	#TODO add note that this Google Project is where the compute will run, the billing account is what will be charged when user runs the sample script, or when they use the config. etc
+    	billing_account_id = raw_input("(IDs are case-sensitive and will look similar to this: 002481-B7351F-CD111E): ")
+    	print "You have selected this Billing Account: %s" % billing_account_id
+
+    	# Project name with datetime stamp (minute and second) and user's email address
+    	# DO NOT PUT AN UNDERSCORE '_' IN THE NAME, it cannot be longer than 30 characters, nor can it have "google"
+    	global project_name
+    	project_name = "cromwell-" + "%s" % (check_output(['gcloud', 'config', 'get-value', 'core/account']).rstrip().partition("@")[0]) + datetime.datetime.now().strftime("-%M-%S")
+    	create_google_project(project_name, billing_account_id)
+
+# Create a google project for the user
+def create_google_project(project_name, billing_account_id):
+	# Create google project
+	print "Creating Google project..."
+	body = {'project_id': '%s' % project_name, 'name': '%s' % project_name, 'labels': None}
+	crm.projects().create(body=body).execute()
+	# Sleep is necessary to give time to create project
+	#TODO decrease time?
+	time.sleep(10) 
+	# Link new project to billing account
+	print "Linking project to your billing account..."
+	enable_billing_account(billing_account_id, project_name)
+	return project_name
+
+# Link the newly created Google project to the user's chosen billing account
+def enable_billing_account(billing_account_id, project_name):
+	body = {"project_id": "%s" % project_name, "billing_account_name": "billingAccounts/%s" % billing_account_id, "billing_enabled": "True"}
+	params = {"name": "projects/%s" % project_name, "body": body}
+	
+	# Enable billing account
+	billing.projects().updateBillingInfo(**params).execute()
+	print "Project created successfully. View your new project here: https://console.cloud.google.com/home/dashboard?project=%s" % project_name
+	create_google_bucket(project_name)
+
+def create_google_bucket(project_name):
+	#TODO: change bucket name to not include datetime
+	#TODO: handle exception if existing bucket
+	print "\nStep (3): Create a Google bucket, starting now..."
+	global bucket_name
+	bucket_name = "%s" % project_name + "-executions"
+	storage_client.create_bucket(bucket_name)
+	print "Bucket created successfully. View your new bucket here: https://console.cloud.google.com/storage/browser/%s" % bucket_name
+	return bucket_name
+
+#TODO: ask for dockerhub credentials if they are going to use private dockers
+
+def create_config():
+	#TODO: make `home` a global variable to remove duplication from inital check for existing config file
+	print "\nStep (4): Create configuration file, starting now..."
+	home = os.path.expanduser("~")
+	config = open(home + "/.google_cromwell.config","w+")
```

Comment:
use the with pattern here: http://www.pythonforbeginners.com/files/with-statement-in-python